### PR TITLE
[AMORO-3677] Fill catalogProperties in creating catalog rather than building

### LINKED ISSUE #3677: [Bug]: hive catalog clients properties is overridden by the default value
### What happened?

- Before creation

Image

- After creation
Image

### Affects Versions

master

### What table formats are you seeing the problem on?

Iceberg

### What engines are you seeing the problem on?

AMS

### How to reproduce

1. create a external hive catalog and set clients to 15 in Properties.
2. save，clients is actually set to 20

### Relevant log output

```shell

```

### Anything else

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MarigWeizhi**
## Why are the changes needed?

Close #3677.

## Brief change log

- Set clients for hive catalog if not exists

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? ( no)
- If yes, how is the feature documented? ( not documented)

### CODE REVIEW

File: amoro-ams/src/main/java/org/apache/amoro/server/catalog/CatalogBuilder.java [hunk 1]
```diff
@@ -90,9 +90,11 @@ public static ServerCatalog buildServerCatalog(
       case CATALOG_TYPE_HIVE:
         String amsUri = getAmsURI(serverConfiguration);
         catalogMeta.getCatalogProperties().put(CatalogMetaProperties.AMS_URI, amsUri);
-        catalogMeta
-            .getCatalogProperties()
-            .put(CLIENT_POOL_SIZE, String.valueOf(CLIENT_POOL_SIZE_DEFAULT));
+        if (!catalogMeta.getCatalogProperties().containsKey(CLIENT_POOL_SIZE)) {
```
Comment (klion26) [inline amoro-ams/src/main/java/org/apache/amoro/server/catalog/CatalogBuilder.java]: If the logic here overrides the value the user set, does that mean we can safely remove this logic here?

Comment (klion26) [reply amoro-ams/src/main/java/org/apache/amoro/server/catalog/CatalogBuilder.java]: Could we add some tests to cover this logic?

File: amoro-ams/src/main/java/org/apache/amoro/server/catalog/CatalogBuilder.java [hunk 2]
```diff
@@ -90,9 +90,11 @@ public static ServerCatalog buildServerCatalog(
       case CATALOG_TYPE_HIVE:
         String amsUri = getAmsURI(serverConfiguration);
         catalogMeta.getCatalogProperties().put(CatalogMetaProperties.AMS_URI, amsUri);
-        catalogMeta
-            .getCatalogProperties()
-            .put(CLIENT_POOL_SIZE, String.valueOf(CLIENT_POOL_SIZE_DEFAULT));
```
Comment (zhongqishang) [inline amoro-ams/src/main/java/org/apache/amoro/server/catalog/CatalogBuilder.java:95]: Maybe just `put` -> `putIfAbsent`.

**Comment by MarigWeizhi** [comment]
@klion26 @zhongqishang Thanks for the review, I updated the code and added a test.

**Comment by klion26** [comment]
@MarigWeizhi Thanks for the follow-up, IIUC, we'll merge the property when constructing the catalog [here](https://github.com/apache/amoro/blob/02a2c1fb2c940a84c310e4e336bab0e25d0b20fa/amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/CatalogController.java#L416). Could you please help verify that it runs as  expected after we remove the logic here(line 93-95 of `amoro-ams/src/main/java/org/apache/amoro/server/catalog/CatalogBuilder.java`), which will update the property.

**Comment by MarigWeizhi** [comment]
@klion26 Thank you for your reply.   
I think these three lines of code `CatalogBuilder.java (lines 93–95)` is to allow users to see the default value of `clients` in the Web UI even when it is not explicitly set, making it easier to view and modify.

I tested two case after removing these three lines in `CatalogBuilder.java` (lines 93–95):

1. **When creating a catalog without configuring `clients`**:
   - The catalog does **not** include the `clients` parameter.
   - As a result, the `clients` field is **not visible** in the Web UI.
   - Eventually, the default value of `20` is used at runtime, as defined here:  
     [CachedHiveClientPool.java#L54](https://github.com/apache/amoro/blob/master/amoro-common/src/main/java/org/apache/amoro/hive/CachedHiveClientPool.java#L54)

   Screenshot 1

2. **When creating a catalog with `clients=15`**:
   - The catalog correctly uses the configured value of `15`.
   - The `clients` parameter is displayed and editable in the Web UI.

   Screenshot 2

**Comment by klion26** [comment]
@MarigWeizhi Thanks for the verification. Seems the result is as expected?
1. If the property was not set, then Amoro will use the default parameter, and the value was not displayed in the ui is as expected, as the user hasn't set it, but maybe we can log the value we used in the log or somewhere
2. If a user sets the property, then Amoro will use the user-set property

I prefer to handle the configuration in a single place, so that we can maintain it easily. Please let me know what you think about this? thanks.

**Comment by MarigWeizhi** [comment]
@klion26 Thank you for your suggestion. I think it's a good idea to control the parameters in one place. I have updated the code and title of this PR.

**Comment by zhoujinsong** [comment]
It seems this PR just rolls back the change produced by #3590.

How can we make sure the default client is 20 now?

**Comment by MarigWeizhi** [comment]
> It seems this PR just rolls back the change produced by #3590.
> 
> How can we make sure the default client is 20 now?

@zhoujinsong 
This PR rolled back part of #3590, to avoid setting default values in two places.  The default value remains 20.  
If the user has not set the clients, a default value will be set at [CachedHiveClientPool.java#L54](https://github.com/apache/amoro/blob/master/amoro-common/src/main/java/org/apache/amoro/hive/CachedHiveClientPool.java#L54)
and default value is still 20 [CatalogMetaProperties.java#L67](https://github.com/apache/amoro/blob/master/amoro-common/src/main/java/org/apache/amoro/properties/CatalogMetaProperties.java#L67)

**Comment by zhoujinsong** [comment]
It seems only the MixedHiveCatalog and HudiHiveCatalog are dependent on [CachedHiveClientPool.java#L54](https://github.com/apache/amoro/blob/master/amoro-common/src/main/java/org/apache/amoro/hive/CachedHiveClientPool.java#L54).

I am afraid the Iceberg Hive Catalog does not work with it.

**Comment by zhoujinsong** [comment]
Besides, we may add the `clients` properties to the catalog meta in the creating catalog process rather than the build catalog method.

It may resolve the issue.

**Comment by MarigWeizhi** [comment]
> Besides, we may add the `clients` properties to the catalog meta in the creating catalog process rather than the build catalog method.
> 
> It may resolve the issue.

@zhoujinsong 
Thank you for your suggestion. I updated the code, I set amsUri and clients when creating catalog . Please take a look when you have time.

**Review by zhoujinsong — APPROVED** [review]
LGTM.
